### ISSUE TITLE
[M] A&R -> Detect null/missing values

### ISSUE THREAD

**Issue by Dianadec** [bcgov/bcparks-ar-admin#351]
#### Description:

Currently no way to detect in system if / how much expected data is missing and where. Variance check only runs on entered values.  

June 12/24 mtg notes: 
- Validating data at the end of the season take a loot of time – lots of back and forth emails. So if running a variance check on null values (-100 variance)  
- Would help park ops, regional staff, and Sally’s team 

#### Acceptance Criteria:

Given
When
Then 

Given
When
Then

#### Development Checklist:

- [ ] ...
- [ ] ...
- [ ] ...

#### Dependencies

- Blocked by #356 
- Blocking

**Wireframe(s):**

**Workflows:**

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by Dianadec** [comment]
Questions for Refinement: discuss how the variance check only runs on entered values, and no way in A&R to see if/how much expected data is missing/where. How can this be addressed?

**Comment by Dianadec** [comment]
Refinement summary:
Want to find data; not change calculation
Wants to check for missing values from past years
Flag for missing

--> Need to create a design to show what this would look like. Eg: a report? or the Review Data screen

Outstanding questions: 
- how to filter so we don't show absolutely everything?
- Could we sort this by region or section? (tbc by team)
- 
todo: create another ticket for the design piece; bring up next refinement; aim to bring in both tix next sprint.

**Comment by nicolebottoms** [comment]
option to add a filter for missing data potentially? and then have the null value as a part of the variance report export? part of a larger thing to flag, the variance report does have a few bugs.
![snip](https://github.com/user-attachments/assets/c22dd280-a1d1-47c3-9667-349b04153753)

**Comment by Dianadec** [comment]
@nicolebottoms are these the bugs you mentioned/related tix? any others? the full list of A&R bugs I'm seeing is in this view: https://github.com/orgs/bcgov/projects/49/views/9?filterQuery=repo%3A%22bcgov%2Fbcparks-ar-admin%22%2C%22bcgov%2Fbcparks-ar-api%22+bug
#277 
#279 
#348

**Comment by nicolebottoms** [comment]
#277 separate from the variance export issue, but that is tied to the null value request.
#279 no idea about this one honestly, it looks to be more of a design and 'make it pretty' thing.
#348 also unrelated, this is for the uploading and exporting of historical data, i think its captured in another ticket but i couldn't see it.

the variance report exports each variance into one cell, i've attached a snip below:
![image](https://github.com/user-attachments/assets/8cfc187b-3946-4304-aebf-0263ebd706ff)
in order to make this more useable and easier to interpret, each variance should be in its own column (red and blue circles). to make this readable i have to manipulate the data to separate the data in the cell, which takes time, and for other users may not be possible depending on their excel skills. i have no idea the best way to make that possible, whether it is easier for each variance to be on its own line, or if it should be in separate columns.

**Comment by Dianadec** [comment]
Thanks @nicolebottoms ! I'll add that bug as a separate ticket

**Comment by Dianadec** [comment]
> option to add a filter for missing data potentially? and then have the null value as a part of the variance report export? part of a larger thing to flag, the variance report does have a few bugs with how the data is displayed in the report ![snip](https://private-user-images.githubusercontent.com/172624036/351427992-c22dd280-a1d1-47c3-9667-349b04153753.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIzNTk2NTcsIm5iZiI6MTcyMjM1OTM1NywicGF0aCI6Ii8xNzI2MjQwMzYvMzUxNDI3OTkyLWMyMmRkMjgwLWExZDEtNDdjMy05NjY3LTM0OWIwNDE1Mzc1My5QTkc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDczMFQxNzA5MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kM2E5YmQwMWY4Mjk2YmVmMDM4MGRhNDg1YjNiN2E3MTI2YjZjNGQ0OWJlYjViNDAwMjk1Yzk5ZTA0MTZiYmVjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ohAF_gW27gmWuaflNWu1dZJf0lahxfIRqEErsKy4xaE)

Cam said this wouldn't be the right spot.


We need to find entries that have never been entered. Provide some UI way of finding that.

Next steps: dev/designer work together on #356

**Comment by Dianadec** [comment]
Designs are in the midst of approvals - we can pull this out of blocked once this has been updated with the design. cc @meyerdarcie

**Comment by davidclaveau** [comment]
@manuji this feature is in DEV now and ready for review!

**Comment by manuji** [comment]
Hi @davidclaveau , this is the exported report I get for "Export all", do you know if this is what to expect? I don't see any park names or any useful information in it. I am a bit lost. 

[A&R_Missing_Report (4).csv](https://github.com/user-attachments/files/17008471/A.R_Missing_Report.4.csv)

**Comment by davidclaveau** [comment]
@manuji this would be expected behaviour (at least for TEST's dataset) when there isn't any "missing data" to export for any park during the chosen Fiscal Year.

The Export All checkbox is meant to export all the parks _but only_ if they trigger the "missing data" logic in the backend - somewhat misleading as it won't print out _everything_ as a default setting.

There's one park with a missing dataset for March 2022 - April 2023, and that'll be Garibaldi Park - Cheakamus - Day Use.

![Image](https://github.com/user-attachments/assets/9aba49a4-e46e-4965-96a9-5428533f6994)

And it does print out (at the very end of the spreadsheet) the 2022 dataset and 2023's missing dataset.

[A&R_Missing_Report (2).csv](https://github.com/user-attachments/files/17030541/A.R_Missing_Report.2.csv)

I added another dataset where Strathcona - Buttle Lake - Day Use January 2022 has data, but January 2023 has "0" data, so now that will appear in the report for Export All.

[A&R_Missing_Report (3).csv](https://github.com/user-attachments/files/17030581/A.R_Missing_Report.3.csv)

Let me know if that loads for you and if it looks good for that Fiscal Year! Hopefully it's not a different issue than that 😅

**Comment by manuji** [comment]
Tested on TEST: **Passed**
* Verified that the missing records will be flagged and displayed in the exported report, when a park/subarea and facility has information for a specific month in a previous year but entered with zero for another year. 
eg: For Strathcona Park - BC, Buttle Lake sub area, Day-use facility, there is information for 2023-August, but for same park, subarea, and facility for 2024- August it has entered with a zero. 

[A&R_Missing_Report (10).csv](https://github.com/user-attachments/files/17036321/A.R_Missing_Report.10.csv)

* passing this to user acceptance testing on the test environment before we move to PROD deployment: https://d3ahrzu0mpndj3.cloudfront.net/export-reports

@sallyhaynes or @ndelraye , could you please review this feature on test environment and give us your feedback before we deploy to PROD environment? 

Thanks in advance!

**Comment by manuji** [comment]
Deployed to PROD after getting UAT feedback here: https://teams.microsoft.com/l/message/19:8397f045482c444fadac0b25c842d605@thread.tacv2/1726676244397?tenantId=6fdb5200-3d0d-4a8a-b036-d3685e359adc&groupId=cb2c4bec-527d-4097-80b2-6e40e98d36bf&parentMessageId=1722552298036&teamName=Parks%20%26%20Recreation%20Transformation%20(Open%20X-Gov)&channelName=Attendance%20and%20Revenue&createdTime=1726676244397

The nice to have requests will be addressed in separate tickets in future.

**Comment by davidclaveau** [comment]
Nice to have tickets here:

- A&R API [374](https://github.com/bcgov/bcparks-ar-api/issues/374)
- A&R Admin [384](https://github.com/bcgov/bcparks-ar-admin/issues/384)

**Comment by JLWade** [comment]
looks good - moving to done
